numpy/lib/_histograms_impl.py

```diff
@@ -725,7 +726,9 @@ def histogram(a, bins=10, range=None, density=None, weights=None):
     -------
     hist : array
         The values of the histogram. See `density` and `weights` for a
-        description of the possible semantics.
+        description of the possible semantics.  If `weights` are given,
+        ``hist.dtype`` will be taken from `weights`, otherwise it will
+        hold large integers.
```

Comment:
```suggestion
        ``hist.dtype`` will be taken from `weights`.
```

Either delete this or rephrase a bit, "hold large integers" is ambiguous. I also think the explanation you've added in the `weights` description is sufficient, no need to repeat with different wording it here.

Reply:
I looked at it again and just followed your suggestion / pushed an updated branch. (For a clean history, that second commit could be a "fixup" commit squashed into the first, but I don't know how the numpy merge process looks like.)